### PR TITLE
Improve multiple strings in en-US locale

### PR DESCRIPTION
**Pull request by n0toose**
I kept sending pull requests that consisted of one-line changes. It's time to
settle this once and for all. (Maybe.)

- Explain Gitea behavior and the consequences of each
  setting better, so that the user does not have to consult
  the docs.
- Do not use different spellings of identical terms
  interchangeably, e.g. `e-mail` and `email`.
- Use more conventional terms to describe the same things,
  e.g. `Confirm Password` instead of `Re-Type Password`.
- Introduces additional clarification for Mirror Settings
- Small adjustments in test
- This is a cry for help.
- Grammar and spelling consistencies for en-US locale
  (e.g. cancelled -> canceled)
- Introduce tooltip improvements.

### CODE REVIEW

**Comment by n0toose** [comment]
I understand that all of these changes at once make for a stressful review process, but I figured that incremental fixes (as I have done so far) probably take up more resources on both ends here. I tried to go for being consistent; I think my biggest hesitation here involved the way I appended the word "successfully".

I'm a details person; I hope that this will make Gitea look more polished. I will not be available in the next few days (as in, it might take at least 12 hours for me to see replies), so feel free to merge any suggestions without my permission.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: I think `Appearance` is better and it is widely used.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: IIRC, the word "create" is used as: "create something" or "something is created"

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
report_message = If you believe that this is a Gitea bug, please search for issues on <a href="https://github.com/go-gitea/gitea/issues" target="_blank">GitHub</a> or open a new issue if necessary.
```

---

Oh, now is also a good time to get rid of the hardcoded link:

```suggestion
report_message = If you believe that this is a Gitea bug, please search for issues on <a href="%[1]s" target="_blank">GitHub</a> or open a new issue if necessary.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
biography_placeholder = Tell us a little bit about yourself! (You can use Markdown)
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
authorized_oauth2_applications_description = You have granted access to your personal Gitea account to these third party applications. Please revoke access for applications you no longer need.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
twofa_scratch_token_regenerated = Your scratch token is now %s. Store it in a safe place, it will never be shown again.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
archive.title = This repo is archived. You can view files and clone it, but cannot push or open issues or pull requests.
archive.title_date = This repository has been archived on %s. You can view files and clone it, but cannot push or open issues or pull requests.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
issues.due_date_not_writer = "You need write access to this repository in order to update the due date of an issue."
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: Did you use a regex?
```suggestion
pulls.no_merge_access = You are not authorized to merge this pull request.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
settings.archive.text = Archiving the repo will make it entirely read-only. It will be hidden from the dashboard. Nobody (not even you!) will be able to make new commits, or open any issues or pull requests.
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: Why titlecase all of a sudden?
```suggestion
settings.unarchive.button = Unarchive repo
settings.unarchive.header = Unarchive this repo
```

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]:
```suggestion
release.deletion_desc = Deleting a release only removes it from Gitea. It will not affect the Git tag, the contents of your repository or its history. Continue?
```

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: Presumably unsuccessfully, I tried to go over it once again after I noticed that I messed some things up but I did not correct everything back, it seems.

**Comment by n0toose** [comment]
Thanks for the additional recommendations and sorry for that regex fail, really goes on to show how this is super subjective work.

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: Of course, if we extract out the link, we need to fix it as well in
```go
templates/status/500.tmpl
45:                                     {{if .SignedUser.IsAdmin}}<p>{{.locale.Tr "error.report_message" | Str2html}}</p>{{end}}
```

Comment (n0toose) [reply options/locale/locale_en-US.ini]: I am not sure how I can approach this best.

Comment (delvh) [reply options/locale/locale_en-US.ini]: I'd guess by replacing the line with
```tmpl
                                     {{if .SignedUser.IsAdmin}}<p>{{.locale.Tr "error.report_message" "https://github.com/go-gitea/gitea/issues" | Str2html}}</p>{{end}}
```

Comment (n0toose) [reply options/locale/locale_en-US.ini]: Sorry, overcomplicated things and started thinking about whether this should be a config option or whatever. Yeah, makes sense.

**Review by delvh — APPROVED** [review]

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: It seems strange that the URL comes from template string but the link text is hard-coded here.

The GitHub should always mean "https://github.com/go-gitea/gitea/issues", right?

I would suggest don't change it at the moment.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: The sentence reads strange, because IMO:

1. Not only "Git operations made using the web interface", but all related "Git operations"
2. Not only notifications, but I think there are also emails for verification or password recovery?

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini:859]: IIRC "revoke" is a "vt": revoke something or something is revoked

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: Actually, either is fine:

https://www.grammarly.com/blog/canceled-vs-cancelled/

> Canceled or cancelled is the past tense of the verb to cancel. Both spellings are correct; Americans favor canceled (one l), while cancelled (two l's) is preferred in British English and other dialects. Canceling/cancelling and canceler/canceller follow the same pattern.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini:2475]:
```suggestion
branch.delete_desc = Deleting a branch is permanent. Although the deleted branch may continue existing for a short time before it actually gets purged, it CANNOT be undone in most cases. Continue?
```

Like this?

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: > Not only "Git operations made using the web interface", but all related "Git operations"

If I make a commit using a secondary email that is not primary, my primary email address will not be used. Unless if you mean pull requests.

> Not only notifications, but I think there are also emails for verification or password recovery?

Good point.

Comment (wxiaoguang) [reply options/locale/locale_en-US.ini]: > > Not only "Git operations made using the web interface", but all related "Git operations"
> 
> If I make a commit using a secondary email that is not primary, my primary email address will not be used.

I see your point and I misunderstood it before. I think it means "this email will be used as the committer email for web Git operations"


```suggestion
email_desc = Your primary email address will be used for notifications and as the committer email for web Git operations.
```

Like this?

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: This is `en-US`, I'll replace `cancelled` in other places as well.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini:2475]: That's a bad grammar mistake (I confuse it because of `cease to exist`, its opposite, being written that way), I'll commit that.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: I also made GitHub a parameter, should I also make `Gitea` a parameter while I'm at it? I presume that you do not provide support for soft-forks like Forgejo, so it could make sense.

Of course, I'm willing to undo that change completely if possible and discuss this in a separate PR - Causing tension is not my intention (oh hey, that rhymes!).

Comment (wxiaoguang) [reply options/locale/locale_en-US.ini]: Never mind, either is fine to me as long as it makes sense. Thank you.

---

ps: the template code hasn't been updated yet.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini:859]: I'm trying make this string consistent with other messages like `pulls.merged_success`, `pulls.required_status_check_failed`, `pulls.update_branch_success`, `settings.transfer.success`, `settings.archive.success`.

These examples are a bit more detailed, but I found it better than changing the string to something like "You have successfully revoked access to this application." or "This application's access was revoked".

I also thing that `you've` should be replaced in most cases, I'll make another commit for that.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: I'll change it to web-based, I think it's very much clear as day and I don't think it will confuse more advanced users that would presumably use the `tea` CLI tool later on.

Comment (n0toose) [reply options/locale/locale_en-US.ini]: Keep in mind, there's many users (myself included, 2 months ago) that do not get that hiding the email address results in a noreply email address being used - this should be mentioned somewhere. It's why I made that change with the padlock - the recommendation to add a tooltip that someone made to the PR (as well as a previous commit that would never make emails accessible to everyone on the Internet) has vastly improved the situation, as people are less likely to hide it. Maybe a tooltip over the "Hide Email Address" option could help later.

Comment (n0toose) [reply options/locale/locale_en-US.ini]: 32ce04ce28078464d5248162e3ab479d1ba23394 makes the text much longer, but I think that the consequences of email-related settings are very well defined. I hope that we won't have to revisit this later down the line.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: I think we can use Wikipedia's example if we'd like to improve it:  https://en.wikipedia.org/wiki/OpenID

```suggestion
openid_signin_desc = Enter your OpenID URI. For example: alice.openid.example.org or https://openid.example.org/alice.
```

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [reply options/locale/locale_en-US.ini]: The sentence `will be used for .... and, provided that it is not hidden, web-based Git operations` reads a little difficult to me. But I am not a native English speaker, so I can't comment on it (aka, no more option from my side).

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: Good idea. I think I'm done here for the rest of this decade (or until I hear about more users "doing things wrong").

**Review by denyskon — APPROVED** [review]
Thank you for these changes!

**Comment by denyskon** [comment]
@n0toose Please fix the test

**Comment by n0toose** [comment]
> Please fix the test

Sorry, I mistakenly assumed that my changes were unrelated to the test fail. This should be fixed now (hopefully).

**Comment by n0toose** [comment]
### Additional Screenshots

image

image

The second change was made as a follow-up to a previous PR of mine, https://github.com/go-gitea/gitea/pull/24433, after a user reported that the lack of `settings.mirror_settings.docs.more_information_if_disabled` was confusing, as immediately referring to the title "How do I mirror repositories?" without any given context gave the false impression that you could, indeed, still mirror repositories, even after the sentence saying that the administrator has disabled this functionality. This new string is only shown when either a pull or a push mirror is disabled as some sort of a bridge in order to signify to the user that the information contained in the following document is more generalized: This may not be obvious to users in public instances. (So, I did not have to change the substance, as in, the title of the document.)

**Comment by n0toose** [comment]
I don't think I can get myself to stop expanding the scope of this PR.

image

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: I don't think I can come up with a better combination of being accurate, easy to read and being transparent. I'm not a native speaker either, and we have to consider that many will use the software in this locale despite not being native speakers, so I'm definitely open to recommendations.

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini:1962]: I would have expected a link here?

**Comment by wxiaoguang** [comment]
This review hasn't been addressed yet: https://github.com/go-gitea/gitea/pull/26213/files#r1278291138

`Access revoked successfully.`

> IIRC "revoke" is a "vt": revoke something or something is revoked

If it's sure that it is correct English syntax, then it can be marked as resolved.

File: options/locale/locale_en-US.ini
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: "disappear" reads like "the secret will be unavailable / will not exist" (IIRC "disappear" means "not exist")

Maybe it could be: "The secret is shown only once on this page" ?

Comment (n0toose) [reply options/locale/locale_en-US.ini]: I also thought of that, but thought that `Please save your secret.` essentially fixes this ambiguity.

Comment (n0toose) [reply options/locale/locale_en-US.ini]: Made new commit, marking this as resolved.

File: options/locale/locale_en-US.ini
Comment (lunny) [inline options/locale/locale_en-US.ini]: Maybe it's better to change the key to avoid other languages rendering error?

Comment (n0toose) [reply options/locale/locale_en-US.ini]: Hi, thanks for the feedback, but I'm really not sure as to how I can address this. Do you want me to revert my changes here?

**Comment by silverwind** [comment]
Ready to merge?

**Comment by n0toose** [comment]
From my side, I think I am done - I am not sure what to do regarding [that](https://github.com/go-gitea/gitea/pull/26213#discussion_r1278665394) though.

**Comment by n0toose** [comment]
Alternative proposal: Revert that change as @lunny mentioned, and then make changes everywhere so that parameters will be used instead of URLs and tags soon after the release of a new version so that translators have enough time to update the strings accordingly.

File: options/locale/locale_en-US.ini
Comment (silverwind) [reply options/locale/locale_en-US.ini]: He meant that outdated translations which do not have the arguments may error because we pass more arguments in than the translation accepts, but I think this is something the translation function could gracefully handle.

Too much arguments should be gracefully ignored, only too few arguments is a real problem.

Comment (silverwind) [reply options/locale/locale_en-US.ini]: Actually, it may not even be an issue because as far as I understand, the change will invalidate all existing translations, so they should fallback to english until translated again, and no further changes are necessary here.

**Comment by silverwind** [comment]
I'm 95% sure it's not an issue because changing a translation in english will invalidate all existing translations and it will fall back to english until translated again.

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I think lunny is right (I misunderstood crowdin too)

For example, the content of `milestones.completeness` in `en_US` has changed, it gets a `<strong>` tag, but in other languages, they all still use old translation, and crowdin didn't show re-translation for this case.

**Comment by lunny** [comment]
> I'm 95% sure it's not an issue because changing a translation in english will invalidate all existing translations and it will fall back to english until translated again.

If those are not invalid, they maybe missed to translate? I don't know whether crowdin will think that key needs to be updated for all translations. But change the key will ask other translations to change otherwise it will not work well.

**Comment by silverwind** [comment]
This translation system sounds so broken if a change in the primary languages does not invalidate all existing translations. I think we ought to write some tooling for this that automatically deletes any changed keys in other translation files.

**Comment by n0toose** [comment]
Asked all three of you for reviews because I either want a final confirmation that I addressed some comments correctly or because there have been a LOT of changes ever since the "Approval". I apologize in advance for the duplicated effort.

I tried my best not to address the parameter problems for the moment, as that seems like an entirely separate issue that is out of my reach as a contributor.

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: That's not true, there are also other reasons why an account cannot sign in, i.e. it is a bot account or belongs to an organization.

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: I think the old one is better

File: options/locale/locale_en-US.ini
Comment (delvh) [inline options/locale/locale_en-US.ini]: Isn't `configuration` a word like `sheep` or `fish` whose plural is its singular?

**Review by wxiaoguang — COMMENTED** [review]
Thank you for inviting me to review, while I am not an English native speaker, I think people understand English better could help.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini:1959]: See https://github.com/go-gitea/gitea/pull/26213/files#diff-dda0bd64abe99b2fc96238e95a74eddf6b89c1210a07130f1d6bedf7528cf461R1961.

Comment (n0toose) [reply options/locale/locale_en-US.ini:1962]: No, `settings.mirror_settings.docs.doc_link_title` (with a URL) follows.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: ACK

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: Me too.

File: options/locale/locale_en-US.ini
Comment (n0toose) [reply options/locale/locale_en-US.ini]: This is correct. Bear in mind that I work on this PR very late at night so that I can actually handle the mundane-ness of it, which is good for progress but still results in many little errors. I'll fix this.

Comment (n0toose) [reply options/locale/locale_en-US.ini]: Hm,
- https://www.merriam-webster.com/dictionary/configuration (`configurations` example used)
- https://www.wordhippo.com/what-is/the-plural-of/configuration.html

I think we are both right, but it may be the case that using plural in the first place makes the idea that is meant to be conveyed here too inaccurate.

Comment (n0toose) [reply options/locale/locale_en-US.ini]: I tried something else here: https://github.com/go-gitea/gitea/pull/26213/commits/faf68dce4f2d5ec3caa731aedc2125ec6261207f

**Comment by silverwind** [comment]
> > I'm 95% sure it's not an issue because changing a translation in english will invalidate all existing translations and it will fall back to english until translated again.
> 
> If those are not invalid, they maybe missed to translate? I don't know whether crowdin will detec that key have been updated and ask all transations to update the key.
> 
> But changing the key can do that otherwise it will not work well.

I think it's default behaviour of crowdin to invalidate existing translation when the primary value changes. This is [this option](https://developer.crowdin.com/configuration-file/#changed-strings-update) which by default is not set, so I assume we don't set it and I recommend to never set it.

**Review by silverwind — APPROVED** [review]

**Comment by denyskon** [comment]
Can we merge this? We could infinitely continue adding stuff here if we wanted, but I think we should merge

**Pull request merged**
